src/ComplaintBox.tsx

```diff
@@ -0,0 +1,193 @@
+import { useEffect } from "react";
+import { observer } from "mobx-react";
+import { Box, Container, Stack, Typography } from "@mui/material";
+import { useRootStore } from "./App";
+import {
+  Button,
+  DropdownInputField,
+  TextAreaInputField,
+  TextInputField,
+  useSpacing,
+  useTypography,
+} from "@surya-digital/leo-reactjs-material-ui";
+import { useTranslation } from "react-i18next";
+// import { DateTime } from "luxon";
+
+export const ComplaintBox = observer((): React.ReactElement => {
+  const {
+    name,
+    setName,
+    mail,
+    selectedType,
+    createdAt,
+    setSelectedType,
+    setMail,
+    complaint,
+    setComplaint,
+    nameError,
+    mailError,
+    complaintError,
+    users,
+    handleSubmit,
+  } = useRootStore();
+
+  const spacing = useSpacing();
+  const typography = useTypography();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    let retrievedName = window.localStorage.getItem("name");
+    if (retrievedName) setName(retrievedName);
+    let retrievedMail = window.localStorage.getItem("mail");
+    if (retrievedMail) setMail(retrievedMail);
+    let retrievedComplaint = window.localStorage.getItem("complaint");
+    if (retrievedComplaint) setComplaint(retrievedComplaint);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    window.localStorage.setItem("name", name);
+  }, [name]);
+  useEffect(() => {
+    window.localStorage.setItem("mail", mail);
+  }, [mail]);
+  useEffect(() => {
+    window.localStorage.setItem("complaint", complaint);
+  }, [complaint]);
+
+  // let dt = DateTime.fromISO(createdAt)
+  //   .setLocale("en-GB")
+  //   .toLocaleString({ month: "long", day: "numeric", year: "numeric" });
+
+  const displayUsers = users.map((user) => (
+    <li key={user.name}>
+      <Typography sx={{ typography: typography.c1 }}>
+        {t("name")}: {user.name}
```

Comment:
This should be done in a single translations

Reply:
Done the change.